### PR TITLE
Tchebytchev 2

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/core/decorator/comp.hpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  //!  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  //! @param f
+  //! An instance of eve::callable
+  //!
+  //! @return
+  //! A @callable performing the same kind of operation but ensuring a higher accuracy.
+  //!
+  //!  @}
+  //================================================================================================
+  [[maybe_unused]] inline constexpr comp_type const comp = {};
```
Comment (jfalcou) [inline include/eve/module/core/decorator/comp.hpp:55]: not fan of the name. comp sounds like comparison.
can't we just ahve compensated ?

File: include/eve/module/core/decorator/kind.hpp [hunk 2]
```diff
@@ -66,4 +66,67 @@ namespace eve
   //!  @}
   //================================================================================================
   inline constexpr q_kind_type const q_kind = {};
+
```
Comment (jfalcou) [inline include/eve/module/core/decorator/kind.hpp:69]: can `kind_1` etc be `kind<N>` ?

File: include/eve/module/core/regular/impl/three_fma.hpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#include <eve/module/core/regular/is_not_finite.hpp>
+#include <eve/module/core/regular/two_prod.hpp>
+#include <eve/module/core/regular/two_add.hpp>
+#include <eve/platform.hpp>
+
+namespace eve::detail
+{
+  template<floating_real_value T>
+  EVE_FORCEINLINE kumi::tuple<T, T, T>
+  three_fma_(EVE_SUPPORTS(cpu_)
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/three_fma.hpp:24]: can't we have a better name ?

File: include/eve/module/polynomial/detail/newton_impl.hpp [hunk 4]
```diff
@@ -57,13 +57,39 @@ namespace eve::detail
...
-    return newton_impl(d, xx, std::begin(rc), std::end(rc), std::begin(rn));
+    using r_t = common_compatible_t<T0, typename R1::value_type, typename R2::value_type>;
+    auto x =  r_t(xx);
+    auto firstc = begin(rc);
+    auto lastc= end(rc);
+    if (firstc == lastc) return r_t(0);
+    else
+    {
+      auto siz = std::distance(firstc, lastc);
+      EVE_ASSERT(siz == inc(std::distance(begin(rn), end(rn))), "number of nodes must be number of coefficients minus 1");
```
Comment (jfalcou) [inline include/eve/module/polynomial/detail/newton_impl.hpp:74]: must be equal to

File: include/eve/module/polynomial/detail/reverse_horner_impl.hpp [hunk 5]
```diff
@@ -23,147 +23,90 @@ namespace eve::detail
   //================================================================================================
   //== N+ 1  parameters (((..(a*x+b)*x+c)*x + ..)..)
   //================================================================================================
-  //==  N = 0
-  template<decorator D,  value T0>
-  EVE_FORCEINLINE constexpr auto reverse_horner_impl(D const &, T0 const &) noexcept
-  {
-    return T0(0);
-  }
+ //  //==  N = 0
```
Comment (jfalcou) [inline include/eve/module/polynomial/detail/reverse_horner_impl.hpp:26]: remove don't comment

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/core/decorator/comp.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/decorator/comp.hpp:55]: ok

File: include/eve/module/core/decorator/kind.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/core/decorator/kind.hpp:69]: yes if I decide to implement more than 2 kinds. Can do it latter/

File: include/eve/module/core/regular/impl/three_fma.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/three_fma.hpp:24]: Its the know name of the algo

File: include/eve/module/polynomial/detail/newton_impl.hpp (hunk 4 above)
Comment (jtlap) [reply include/eve/module/polynomial/detail/newton_impl.hpp:74]: done

File: include/eve/module/polynomial/detail/reverse_horner_impl.hpp (hunk 5 above)
Comment (jtlap) [reply include/eve/module/polynomial/detail/reverse_horner_impl.hpp]: done

**Pull request merged**